### ISSUE TITLE
AttributeError: type object 'IdentifierType' has no attribute '__slots__'

### ISSUE THREAD

**Issue by emezh** [eliben/pycparser#170]
Hello Eli,
I might be misusing the pycparser, so pls let me know if this is indeed the case.
I have a C-header file with few "typedef struct" definitions and i want to convert the structs to jsons.
I tried using examples\c_json.py on my header file and got the error:
    AttributeError: type object 'IdentifierType' has no attribute '__slots__'

Then I tried to run c_json on few c files in examples dir and for few of them i got the same error, e.g.:
```
C:\Users\Eugene\Documents\pycparser-master>python examples\c_json.py tests\c_files\memmgr_with_h.c
Traceback (most recent call last):
  File "examples\c_json.py", line 199, in <module>
    ast_dict = file_to_dict(sys.argv[1])
  File "examples\c_json.py", line 133, in file_to_dict
    return to_dict(ast)
  File "examples\c_json.py", line 113, in to_dict
    result[array_name].append(to_dict(child))
  File "examples\c_json.py", line 115, in to_dict
    result[child_name] = to_dict(child)
  File "examples\c_json.py", line 115, in to_dict
    result[child_name] = to_dict(child)
  File "examples\c_json.py", line 118, in to_dict
    for child_attr in child_attrs_of(klass):
  File "examples\c_json.py", line 64, in __missing__
    ret = self[key] = fn(key)
  File "examples\c_json.py", line 77, in child_attrs_of
    all_attrs = set([i for i in klass.__slots__ if not RE_INTERNAL_ATTR.match(i)])
AttributeError: type object 'IdentifierType' has no attribute '__slots__'

C:\Users\Eugene\Documents\pycparser-master>python examples\c_json.py tests\c_files\example_c_file.c
Traceback (most recent call last):
  File "examples\c_json.py", line 199, in <module>
    ast_dict = file_to_dict(sys.argv[1])
  File "examples\c_json.py", line 133, in file_to_dict
    return to_dict(ast)
  File "examples\c_json.py", line 113, in to_dict
    result[array_name].append(to_dict(child))
  File "examples\c_json.py", line 115, in to_dict
    result[child_name] = to_dict(child)
  File "examples\c_json.py", line 115, in to_dict
    result[child_name] = to_dict(child)
  File "examples\c_json.py", line 118, in to_dict
    for child_attr in child_attrs_of(klass):
  File "examples\c_json.py", line 64, in __missing__
    ret = self[key] = fn(key)
  File "examples\c_json.py", line 77, in child_attrs_of
    all_attrs = set([i for i in klass.__slots__ if not RE_INTERNAL_ATTR.match(i)])
AttributeError: type object 'IdentifierType' has no attribute '__slots__'
```

My system is Win7, 64-bit, Python 2.7.12.

Pls advise.
Thanks,
  Eugene

**Comment by emezh** [comment]
Hmmm... 
Yesterday I installed pycparser using 
`pip install pycparser`
and got the errors I reported few minutes ago.
Just now i tried to (re)install pycparser from the downloaded package using `python setup.py install` and now c_json works ok both for pycparser example C files and my own C header file.

I'm closing the issue.

However, if you have a thought on why "pip install pycparser" didn't install (or update?) the latest version - pls share it.
